ompi/mca/osc/ucx/osc_ucx_passive_target.c

```diff
@@ -124,6 +124,11 @@ int ompi_osc_ucx_lock(int lock_type, int target, int mpi_assert, struct ompi_win
         }
     } else {
         lock->is_nocheck = true;
+        if (lock_type == MPI_LOCK_EXCLUSIVE) {
```

Comment:
This looks good but the same problem exists in `ompi_osc_ucx_lock_all`

Reply:
@devreal Good catch. Will add it there too. Thanks.

Reply:
@devreal I took a look at ompi_osc_ucx_lock_all implementation. Looks like we are not creating any new lock object. Therefore, lock will be NULL inside need_acc_lock(). So we should be good here as if the lock is NULL, need_acc_lock returns true. Please let me know your thoughts.